docs/acceptance-criteria/uacs-tbtc-v2.adoc

```diff
@@ -0,0 +1,291 @@
+
+// The below DKG timeout UACs are describing how timeouts would work if the
+// procedure for submitting DKG result worked as it works currently for Random
+// Beacon (at first one operator can submit, after
+// `resultSubmissionEligibilityDelay` another is added to the list of operators
+// eligible to publish the result, and so on...). This is what initial
+// implementation of a DKG fo ECDSA looks like, but there are ongoing
+// discussions whether this is the best approach. We may end up not having
+// timeouts at all...
+
+| Wallet lifecycle - wallet creation - DKG timeout
+| The DKG submission timeout equals the group size multiplied by the number of
+  blocks for a member to become eligible to submit the DKG result plus the
+  number of blocks covering for the time of DKG generation. The timer should
+  start when the first member becomes eligible (the moment when DKG was
+  requested). The timer gets reset when a valid DKG result challenge is
+  submitted.
+
+| Wallet lifecycle - wallet creation - DKG timeout
+| Anyone can send a transaction reporting DKG timeout.
+
+| Wallet lifecycle - wallet creation - DKG timeout
+| Upon noticing the DKG timeout notification, the DKG contract should determine
+  its validity.
+
+| Wallet lifecycle - wallet creation - DKG timeout
+| When determining the validity of the DKG timeout notification, the following
+  aspects should be taken into account: +
+  - does it reference the existing DKG request? (if not, then notification
+  invalid) +
+  - has the DKG timeout passed? (if not, then notification invalid) +
+  - is the sortition pool in a locked state (if not - meaning somebody already
+  unlocked it - then notification invalid).
+
+| Wallet lifecycle - wallet creation - DKG timeout
+| If the DKG timeout notification was deemed justified, then: +
+  - the pool should be unlocked +
+  - the fixed amount reward (in T) (`sortitionPoolUnlockingReward`) should be
+  sent from the DAO-funded ETH pool to the notifier.
+
+// TODO: add UACs for hearbeats, wallet closure, sweeping, deposits, redemptions, etc
+
+| Governance - group size
+| The number of members selected to the candidate signing group (`group_size`)
+  should NOT be a governable parameter and should be set to `100`.
+
+| Governance - group size
+| The minimum number of members we’re allowed to drop down to during the DKG
+  group formation re-try period (`dkg`) should be a governable parameter and
+  should be initially set to `90`.
+
+| Governance - wallet lifecycle
+| The minimum time since creation of the last wallet after which a new wallet
+  can be created (`wallet_creation_period`) should be a governable parameter and
+  should be initially set to `1 week`.
+
+| Governance - wallet lifecycle
+| The smallest amount of btc a wallet can hold before we attempt to close the
+  wallet and transfer the funds to a randomly selected wallet (`wallet_min_btc`)
+  should be a governable parameter and should be initially set to ...
```

Comment:
How will we treat freashly created wallets, which don't have any deposits on them yet (so their balance is lower than `wallet_min_btc`)? Maybe we should have some period of wallet 'immunity'? So that we wouldn't attemt to close wallets which live for less than `wallet_immunity_period`? If we do that, then we'll need to modify conditions for new wallet creation, so that we'll be able to create a new wallet only if at least 7 days passed since last wallet creation AND all the wallets older than `wallet_immunity_period` have their balances higher than `wallet_min_btc`. We can also consider replacing "AND all the wallets older than `wallet_immunity_period` have their balances higher than `wallet_min_btc`" condition with "AND there is no wallet in the 'emptying' phase".